tests/ducktape/test_transactions.py

```diff
@@ -72,13 +61,17 @@ def _create_transactional_consumer(self, consumer_type, group_id=None):
             return strategy.create_consumer(config_overrides=overrides)
 
     async def _transactional_produce(self, producer, topic: str, values: list[str], partition: int | None = None):
-        """Produce values within an active transaction, polling to drive delivery."""
-        for v in values:
+        """Produce values within an active transaction using async producer."""
+        for i, v in enumerate(values):
             kwargs = {'topic': topic, 'value': v}
             if partition is not None:
                 kwargs['partition'] = partition
-            await producer.produce(**kwargs)
-            await producer.poll(0.0)
+
+            try:
+                await producer.produce(**kwargs)
+            except Exception as e:
+                print(f"ERROR: Failed to queue async message {i}: {e}")
+                raise
```

Comment:
```suggestion
async def _transactional_produce(self, producer, topic: str, values: list[str], partition: int | None = None):
    """Produce values within an active transaction using async producer and return futures."""
    futures = []
    
    for i, v in enumerate(values):
        kwargs = {'topic': topic, 'value': v}
        if partition is not None:
            kwargs['partition'] = partition

        try:
            future = await producer.produce(**kwargs)
            futures.append(future)
        except Exception as e:
            print(f"ERROR: Failed to queue async message {i}: {e}")
            raise
    
    return futures
```

Maybe this function can be updated to return futures. ?